core/domain/user_jobs_continuous.py

```diff
@@ -428,15 +428,28 @@ def reduce(key, stringified_values):
         # Find the average of all average ratings
         ratings = [value['average_rating_for_owned_exp'] for value in values
                    if value.get('average_rating_for_owned_exp')]
+        mr_model = user_models.UserStatsModel.get(key, strict=False)
+
         if len(ratings) != 0:
             average_ratings = (sum(ratings) / float(len(ratings)))
-            user_models.UserStatsModel(
-                id=key,
-                impact_score=user_impact_score,
-                total_plays=total_plays,
-                average_ratings=average_ratings).put()
         else:
-            user_models.UserStatsModel(
-                id=key,
-                impact_score=user_impact_score,
-                total_plays=total_plays).put()
+            average_ratings = None
+
+        if not mr_model:
+            if average_ratings:
```

Comment:
I think a much cleaner way to do this is to follow the `get_or_create()` pattern in other models (for example, core.storage.activity.ActivityReferencesModel). Then, you just call get_or_create, and update the model you get, so you don't need to branch on the `if not mr_model`.